python/runtime/optimize/model_generation.py

```diff
@@ -0,0 +1,624 @@
+# Copyright 2020 The SQLFlow Authors. All rights reserved.
```

Comment:
How about refactoring to:

1. generate a grammar tree (AST) from the SQLFlow expression tokens
2. transverse the grammar tree and generate corresponding expressions for pyomo or optflow.

Can we make the code simpler by using this way?

Reply:
I have thought about this way as well. It may be simpler and more standard. I think that it would be more reasonable if we transform the SQLFlow Go AST to be the final model code, instead of transforming the SQLFlow Go AST to tokens and parse the tokens to another AST in the Python side. I would think about this idea. The related issue is https://github.com/sql-machine-learning/sqlflow/issues/2531 .